File: backend/app/schemas/ticket.py
from datetime import datetime

from pydantic import BaseModel

from app.models.ticket import TicketSource, TicketStatus


class TicketBase(BaseModel):
    title: str
    description: str


class TicketCreate(TicketBase):
    category_id: int | None = None
    priority_id: int | None = None
    source: TicketSource = TicketSource.manual
    requester_email: str | None = None
    requester_name: str | None = None


class TicketUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: TicketStatus | None = None
    category_id: int | None = None
    priority_id: int | None = None
    assigned_agent_name: str | None = None
    agent_response: str | None = None


class TicketResponse(BaseModel):
    id: int
    title: str
    description: str
    status: TicketStatus
    source: TicketSource
    category_id: int | None
    priority_id: int | None
    requester_email: str | None
    requester_name: str | None
    assigned_agent_name: str | None
    agent_response: str | None
    classification_confidence: float | None
    priority_confidence: float | None
    classification_explanation: str | None
    priority_explanation: str | None
    email_sender: str | None
    email_subject: str | None
    email_message_id: str | None
    email_received_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/ticket.ts
export type TicketStatus = 'open' | 'ai_reviewed' | 'pending' | 'resolved' | 'rejected'
export type TicketSource = 'manual'

export const TICKET_STATUS_LABELS: Record<TicketStatus, string> = {
  open: 'Otwarte',
  ai_reviewed: 'Zweryfikowane przez AI',
  pending: 'Oczekujące',
  resolved: 'Rozwiązane',
  rejected: 'Odrzucone',
}

export interface Ticket {
  id: number
  title: string
  description: string
  status: TicketStatus
  source: TicketSource
  category_id: number | null
  priority_id: number | null
  requester_email: string | null
  requester_name: string | null
  assigned_agent_name: string | null
  classification_confidence: number | null
  priority_confidence: number | null
  classification_explanation: string | null
  priority_explanation: string | null
  created_at: string
  updated_at: string
}

export interface TicketCreate {
  title: string
  description: string
  requester_email: string
  requester_name?: string | null
  category_id?: number | null
  priority_id?: number | null
}

export interface TicketUpdate {
  title?: string
  description?: string
  status?: TicketStatus
  category_id?: number | null
  priority_id?: number | null
  assigned_agent_name?: string | null
}
